src/types/simple.rs

```diff
@@ -344,6 +531,23 @@ impl<T: PrimType> TypeRow<T> {
         self.types.len() == 0
     }
 
+    /// Returns whether the row contains only hashable classic data.
+    #[inline(always)]
+    pub fn purely_hashable(&self) -> bool {
+        self.types
+            .iter()
+            .map(PrimType::tag)
+            .all(TypeTag::is_hashable)
+    }
+
+    /// Returns the smallest [TypeTag] that contains all elements of the row
+    pub fn common_class(&self) -> TypeTag {
```

Comment:
This isn't actually used (we could use it for `purely_classical` and `purely_hashable` but that would be marginally less efficient). But it seems handy. If keeping it, we need a better name though. "common_tag" I guess, but that's not great. "union_tag"? "lub" (least upper bound)??

Reply:
`shared_tag`?